main.go

```diff
@@ -841,13 +840,19 @@ func compile(params compileParams) error {
 		}
 		codeGen := newLayoutCodeGen(layout, params.pfile, src)
 		code, err = genCodeLayout(codeGen)
+		if err != nil {
+			return fmt.Errorf("generating code layout: %w", err)
```

Comment:
I would keep it consistent with the new error message for pages below: "generating code for a layout: %w"